\PassOptionsToPackage{unicode=true}{hyperref} % options for packages loaded elsewhere
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provides euro and other symbols
\else % if luatex or xelatex
  \usepackage{unicode-math}
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage[]{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\usepackage{hyperref}
\hypersetup{
            pdftitle={MAGeCK Count Report},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs}
% Fix footnotes in tables (requires footnote package)
\IfFileExists{footnote.sty}{\usepackage{footnote}\makesavenoteenv{longtable}}{}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{0}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

% set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother


\title{MAGeCK Count Report}
\author{}
\date{\vspace{-2.5em}}

\begin{document}
\maketitle

Author: Wei Li, weililab.org

\hypertarget{parameters}{%
\subsection{Parameters}\label{parameters}}

comparison\_name is the prefix of your output file, defined by the
``-n'' parameter in your ``mageck test'' command. The system will look
for the following files to generate this report:

\begin{itemize}
\tightlist
\item
  comparison\_name.countsummary.txt
\item
  comparison\_name.count\_normalized.txt
\item
  comparison\_name.log
\end{itemize}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{# define the comparison_name here; for example,}
\CommentTok{# comparison_name='demo'}
\NormalTok{comparison_name=}\StringTok{'vorinostat'}
\end{Highlighting}
\end{Shaded}

\hypertarget{preprocessing}{%
\subsection{Preprocessing}\label{preprocessing}}

Reading input files. If any of these files are problematic, an error
message will be shown below.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{cstable=}\KeywordTok{read.table}\NormalTok{(count_summary_file,}\DataTypeTok{header =}\NormalTok{ T,}\DataTypeTok{as.is =}\NormalTok{ T)}
\NormalTok{nc_table=}\KeywordTok{read.table}\NormalTok{(normalized_cnt_file,}\DataTypeTok{header =}\NormalTok{ T,}\DataTypeTok{as.is =}\NormalTok{ T)}
\end{Highlighting}
\end{Shaded}

\hypertarget{summary}{%
\subsection{Summary}\label{summary}}

The summary of the count command is as follows.

\begin{longtable}[]{@{}llrrrrrr@{}}
\caption{Count command summary}\tabularnewline
\toprule
File & Label & Reads & Mapped & Percentage & TotalsgRNAs & Zerocounts &
GiniIndex\tabularnewline
\midrule
\endfirsthead
\toprule
File & Label & Reads & Mapped & Percentage & TotalsgRNAs & Zerocounts &
GiniIndex\tabularnewline
\midrule
\endhead
raw\_data/reads\_vorinostat\_screen/2546\_CRISPR\_CM\_DMSO1.fastq &
DMSO1 & 1100545 & 818164 & 0.7434 & 12434 & 195 & 0.1668\tabularnewline
raw\_data/reads\_vorinostat\_screen/2547\_CRISPR\_CM\_DMSO2.fastq &
DMSO2 & 1988853 & 1540808 & 0.7747 & 12434 & 71 & 0.1072\tabularnewline
raw\_data/reads\_vorinostat\_screen/2548\_CRISPR\_CM\_Vorinostat1.fastq
& V1 & 2868072 & 2203415 & 0.7683 & 12434 & 97 & 0.1093\tabularnewline
raw\_data/reads\_vorinostat\_screen/2549\_CRISPR\_CM\_Vorinostat2.fastq
& V2 & 3039906 & 2368991 & 0.7793 & 12434 & 76 & 0.1107\tabularnewline
\bottomrule
\end{longtable}

The meanings of the columns are as follows.

\begin{itemize}
\tightlist
\item
  File: The filename of fastq file;
\item
  Label: Assigned label;
\item
  Reads: The total read count in the fastq file;
\item
  Mapped: Reads that can be mapped to gRNA library;
\item
  Percentage: The percentage of mapped reads;
\item
  TotalsgRNAs: The number of sgRNAs in the library;
\item
  ZeroCounts: The number of sgRNA with 0 read counts;
\item
  GiniIndex: The Gini Index of the read count distribution. Gini index
  can be used to measure the evenness of the read counts, and a smaller
  value means a more even distribution of the read counts.
\end{itemize}

If --day0label and --gmt-file options are provided, the following
metrics will display the degree of negative selections of essential
genes (provided by --gmt-file).

\begin{longtable}[]{@{}llrrrrr@{}}
\caption{Count command summary}\tabularnewline
\toprule
File & Label & NegSelQC & NegSelQCPval & NegSelQCPvalPermutation &
NegSelQCPvalPermutationFDR & NegSelQCGene\tabularnewline
\midrule
\endfirsthead
\toprule
File & Label & NegSelQC & NegSelQCPval & NegSelQCPvalPermutation &
NegSelQCPvalPermutationFDR & NegSelQCGene\tabularnewline
\midrule
\endhead
raw\_data/reads\_vorinostat\_screen/2546\_CRISPR\_CM\_DMSO1.fastq &
DMSO1 & 0 & 1 & 1 & 1 & 0\tabularnewline
raw\_data/reads\_vorinostat\_screen/2547\_CRISPR\_CM\_DMSO2.fastq &
DMSO2 & 0 & 1 & 1 & 1 & 0\tabularnewline
raw\_data/reads\_vorinostat\_screen/2548\_CRISPR\_CM\_Vorinostat1.fastq
& V1 & 0 & 1 & 1 & 1 & 0\tabularnewline
raw\_data/reads\_vorinostat\_screen/2549\_CRISPR\_CM\_Vorinostat2.fastq
& V2 & 0 & 1 & 1 & 1 & 0\tabularnewline
\bottomrule
\end{longtable}

The meanings of the columns are as follows.

\begin{itemize}
\tightlist
\item
  NegSelQC: the enrichment score (ES) of essential genes in the negative
  selection list. The score is calculated using GSEA;
\item
  NegSelQCPval: the associated p value of the enrichment score;
\item
  NegSelQCPvalPermutation: the permutated p value of the enrichment
  score;
\item
  NegSelQCPvalPermutationFDR: the adjusted permutated p value;
\item
  NegSelQCGene: the number of genes used for the analysis.
\end{itemize}

\hypertarget{normalized-read-count-distribution-of-all-samples}{%
\subsection{Normalized read count distribution of all
samples}\label{normalized-read-count-distribution-of-all-samples}}

The following figure shows the distribution of median-normalized read
counts in all samples.

\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-7-1.pdf}

The following figure shows the histogram of median-normalized read
counts in all samples.

\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-8-1.pdf}

\hypertarget{principle-component-analysis}{%
\subsection{Principle Component
Analysis}\label{principle-component-analysis}}

The following figure shows the first 2 principle components (PCs) from
the Principle Component Analysis (PCA), and the percentage of variances
explained by the top PCs.

\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-9-1.pdf}
\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-9-2.pdf}
\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-9-3.pdf}

The variance of the PCs

\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-10-1.pdf}

\hypertarget{sample-clustering}{%
\subsection{Sample clustering}\label{sample-clustering}}

The following figure shows the sample clustering result.

\includegraphics{vorinostat.count_report_files/figure-latex/unnamed-chunk-11-1.pdf}

\end{document}
