---
title: "Targeting EHMT2 sensitizes colorectal cancer to Vorinostat"
author: "Leonhard Valentin Bamberg et al."
date: "`r Sys.Date()`"
output:
  pdf_document: 
    toc: yes
    number_sections: yes
  html_document:
    df_print: paged
bibliography: references.bib
abstract: Epigenetic dysregulation is an important feature of colorectal cancer (CRC). Combining epigenetic drugs with other antineoplastic agents is a promising treatment strategy for advanced cancers. Here, we exploited the concept of synthetic lethality to identify epigenetic targets that act synergistically with histone deacetylase (HDAC) inhibitors to reduce the growth of CRC. We applied a pooled CRISPR-Cas9 screen using a custom sgRNA library directed against 614 epigenetic regulators and discovered that knockout of the euchromatic histone lysine methyltransferases EHMT1 and EHMT2 strongly enhanced the antiproliferative effect of clinically used HDAC inhibitors. Using tissue microarrays from 1066 CRC samples with different tumor stages, we showed that low EHMT2 protein expression is predominantly found in advanced CRC and associated with poor clinical outcome. Co-targeting of HDAC and EHMT1/2 with specific small molecule inhibitors synergistically reduced proliferation of CRC cell lines. Mechanistically, we used a high-throughput Western blot assay to demonstrate that both inhibitors elicited distinct cellular mechanisms to reduce tumor growth, including cell cycle arrest and modulation of autophagy. On the epigenetic level, the compounds increased H3K9 acetylation and reduced H3K9 dimethylation. Finally, we used a panel of patient-derived CRC organoids to show that HDAC and EHMT1/2 inhibition synergistically reduced tumor viability in advanced models of CRC.  
vignette: |
  %\VignetteIndexEntry{Vignette Title} %\VignetteEncoding{UTF-8} %\VignetteEngine{knitr::rmarkdown}
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE, echo=FALSE,warning=F,error=F}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
library(readxl)
library(patchwork)
library(ggrepel)
library(ggsignif)
library(ggpubr)

theme_b110 <- function(){
  theme_classic() +
  theme(
    axis.text = element_text(size = 10), 
    axis.title = element_text(size = 10),
    plot.title = element_text(size = 12,hjust = 0.5,face = "bold"),
    legend.title = element_text(size = 12),
    legend.text = element_text(size = 10),
    legend.position = "bottom"
    )
}

sgi_blue    = "#5087C8"
sgi_yellow1 = "#F2EE35"
sgi_yellow2 = "#FED98E"
b110_grey   = "#808080"
b110_grey_light   = "#909090"
b110_transparent_black = scales::alpha("#000000",0.5)
google_red = "#dd4b39"
google_green = "#0F9D58"
google_yellow = "#F4B400"
google_blue = "#4285F4"

```
\newpage
# Overview

Working Hypothesis: Low EHMT2  expression predicts response to HDACi.

Analysis: Correlating expression level of EHMT2 with response to HDACi

Data sources:
  drug sensitivity data
    GDSC screens [@iorio_landscape_2016]
    PRISM screens [@corsello_discovering_2020]
  expression data
    ArrayExpress (accession number: E-MTAB-3610)
  annotations
    CCLE_DepMap sample_info [@barretina_cancer_2012]

```{r data_read, echo=T,warning=F,error=F,message=F}

#first we read in the cell line annotations

cell_mutant_annotation <- bind_rows(
  list(
    "GDSC1"=read_delim(
      "source_data/COREAD_Genetic_feature_GDSC1_220221_commacorrected.csv",
                       delim = ";",col_types = "cffffcfcc",trim_ws = T),
    "GDSC2"=read_delim(
      "source_data/COREAD_Genetic_feature_GDSC2_220221_commacorrected.csv",
                       delim = ";",col_types = "cffffcfcc",trim_ws = T)),
  .id = "dataset")

#second we read in the actual screening data

screening_data <- bind_rows(
  list(
    "GDSC1"=read_excel("source_data/GDSC1_fitted_dose_response_25Feb20.xlsx"),
    "GDSC2"=read_excel("source_data/GDSC2_fitted_dose_response_25Feb20.xlsx")))

colnames(screening_data) <- tolower(colnames(screening_data))

screening_data <- screening_data %>% rename("cosmic_sample_id"=cosmic_id)

cell_expression_data <- 
  read_delim("source_data/Cell_line_RMA_proc_basalExp_EHMT.txt",delim = "\t")%>%
  gather(cosmic_sample_id,normed_expression,-GENE_SYMBOLS,-GENE_title ) %>%
  extract(cosmic_sample_id,"cosmic_sample_id","DATA.(\\d+)") %>%
  filter(GENE_SYMBOLS %in% c("EHMT2"))

colnames(cell_expression_data) <- tolower(colnames(cell_expression_data)) 

cell_lines_of_interest <- cell_mutant_annotation %>% 
                                filter(gdsc_desc2=="large_intestine") %>% 
                                pull(cosmic_sample_id) %>% 
                                unique()

total_expression_data  <- 
  read_delim("source_data/Cell_line_RMA_proc_basalExp.txt",delim = "\t")%>%
  gather(cosmic_sample_id,normed_expression,-GENE_SYMBOLS,-GENE_title ) %>%
  extract(cosmic_sample_id,"cosmic_sample_id","DATA.(\\d+)")

colnames(total_expression_data) <- tolower(colnames(total_expression_data)) 

```

# Basal gene expression analysis

Hypothesis: EHMT2 expression is variant enough throughout colorectal cancer cell lines to make assumptions on its impact on Vorinostat sensitivity

Thus we test the expression intensity of EHMT2 compared to all other genes and the variance of its expression when compared to the expression variance of all other genes.

EHMT2 expression variance expressed in percent expression variance of the average gene

```{r Expression variance per gene, echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}

#next we calculate the percentage of spread of EHMT2 compared to the total spread of expression values observed in the data
vars<- total_expression_data %>% 
            mutate(cellofinterest=cosmic_sample_id %in% cell_lines_of_interest,
                   geneofinterest=gene_symbols %in% c("EHMT2")) %>%
            filter(cellofinterest) %>% 
            group_by(gene_symbols) %>% 
            summarise(v=var(normed_expression)) %>% 
            ungroup() %>% 
            arrange(desc(v)) %>% 
            mutate(idx=1:n()) 

vars %>%
  mutate(geneofinterest=gene_symbols%in%c("EHMT2")) %>%
  group_by(geneofinterest) %>%
  summarise(m=mean(v)) %>%
  spread(geneofinterest,m) %>%
  mutate(percentVar=`TRUE`*100/`FALSE`)

p1 <- vars %>%
  mutate(geneofinterest=gene_symbols%in%c("EHMT2"))  %>% 
  ggplot(aes(x=idx,y=v)) +
    geom_point(data = vars %>% filter(gene_symbols!="EHMT2"),col="grey") +
    geom_point(data = vars %>% filter(gene_symbols=="EHMT2"),col="orange") +
    geom_label_repel(data = vars %>% filter(gene_symbols=="EHMT2"),
                     aes(label=paste0(gene_symbols," rank: ",idx))) +
    theme_b110() +
    xlab("variability rank") +
    ylab("log10 expression variance") +
    scale_y_log10()

p2 <- total_expression_data %>% 
    mutate(cellofinterest=cosmic_sample_id %in% cell_lines_of_interest,
           geneofinterest=if_else(gene_symbols %in% c("EHMT2"),
                                  "EHMT2",
                                  "ALL")) %>%
    filter(cellofinterest) %>% 
      ggplot(aes(x=geneofinterest,y = normed_expression,fill=geneofinterest)) +
        geom_violin() +
        theme_b110() +
        xlab("gene of interest") +
        ylab("norm. expression") +
        scale_fill_manual(values = c("grey","orange"))

p3 <- p2 + p1

print(p3)

ggsave(filename = "results/GDSC_EHMT_cosmic_expression_variance.pdf",p3,height = 6,width=12)
```

EHMT2 per cent expression range of all genes

```{r Expression range, echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}
#EHMT2 per cent variance of all genes

total_expression_data %>% 
    mutate(cellofinterest=cosmic_sample_id %in% cell_lines_of_interest,
           geneofinterest=gene_symbols %in% c("EHMT2")) %>%
  filter(cellofinterest) %>%
  group_by(geneofinterest) %>%
  summarise(r=abs(range(normed_expression)[1]-range(normed_expression)[2])) %>%
  spread(geneofinterest,r) %>%
  mutate(perc_range=`TRUE`*100/`FALSE`)

```

EHMT2 per cent variance of all genes

```{r Dynamic expression variance, echo=FALSE, message=FALSE, warning=FALSE, paged.print=FALSE}

total_expression_data %>% 
  mutate(cellofinterest=cosmic_sample_id %in% cell_lines_of_interest,
         geneofinterest=gene_symbols %in% c("EHMT2")) %>%
  filter(cellofinterest) %>%
  group_by(geneofinterest) %>%
  summarise( r = var( normed_expression, na.rm = T ) ) %>% 
  spread(geneofinterest,r) %>%
  mutate(perc_range = `TRUE` * 100/ `FALSE`)

```

From this figure and analysis we can conclude that compared to other genes, EHMT2 is rather highly expressed in colorectal cancer cell lines and that the variability of EHMT2 expression between different cell lines is on par with the average variability of expression of any other gene.

```{r data_Wrangling, echo=FALSE,warning=F,error=F,message=F}

# filter the correct data
screening_data_CRC <- 
  screening_data %>% 
    select(cosmic_sample_id,drug_id,drug_name,
           putative_target,ln_ic50,auc,rmse,z_score) %>%
    filter(cosmic_sample_id %in% cell_lines_of_interest)%>%
    mutate(cosmic_sample_id = factor(cosmic_sample_id))


cell_expression_data_CRC <- 
  cell_expression_data %>%
    filter(cosmic_sample_id %in% cell_lines_of_interest) %>% 
    group_by(gene_symbols,cosmic_sample_id) %>%
    summarise(normed_expression=mean(normed_expression,na.rm = T)) %>%
    ungroup() %>%
    spread(gene_symbols,normed_expression)

joined_screening_data_CRC <- 
  screening_data_CRC %>% 
    left_join(cell_expression_data_CRC)
```

\newpage
# EHMT - HDAC relationship (CancerXGene)

First we plot the RMA normalized expression of EHMT2 against the effect Vorinostat have on each cell line. For visualization of trends we also plot a linear models for the data. The slope denotes correlation and the gray confidence interval denotes the certainty on this slope. Further we color the lower and upper quantile of EHMT expression and plot the distribution of points within these quantiles to compare them statistically.


```{r graphics_EHMT_CancerrxGene, echo=FALSE,warning=F,error=F,message=F}

data_to_plot <- joined_screening_data_CRC %>% 
    filter(drug_name %in% c("Vorinostat")) %>%
    mutate(EHMT2q = if_else(EHMT2>quantile(EHMT2,probs = 0.75,na.rm = T),
                            "high", 
                            if_else(EHMT2<quantile(EHMT2,probs = 0.25,na.rm = T),
                                    "low",
                                    "medium"))) %>%
    mutate(EHMT2q=factor(EHMT2q,levels = c("low","medium","high"))) %>%
  drop_na()


a <- data_to_plot %>% 
        ggplot(aes(x = EHMT2,y = auc)) + 
          geom_point(aes(col = EHMT2q)) + 
          geom_smooth(method = "lm")+ 
          stat_cor(method = "pearson", label.x = 4, label.y = 1) + 
          theme_b110()+
          scale_color_manual(values = c(google_blue,
                                        b110_grey_light,
                                        google_red))

give.n <- function(x){
  #return(c(y = max(abs(x))*1.5, label =length(x)))
  return(c(y = 1.1, label =length(x)))
  # experiment with the multiplier to find the perfect position
}


b <- data_to_plot %>%
      filter(EHMT2q != "medium") %>%
      ggplot(aes(x = EHMT2q,y = auc)) + 
        geom_boxplot(aes(fill = EHMT2q)) + 
          facet_wrap(~drug_name,scales = "free_x") +
          theme_b110()+
          scale_fill_manual(values = c(google_blue,google_red)) +
          stat_summary( fun.data = give.n, 
                        geom = "text", 
                       # fun.y = median,
                        position = position_dodge(width = 0.75),fun.ymin = max)+
          geom_signif(na.rm=T,
                      comparisons = list(c("low","high")),
                      test = "t.test",
                      test.args = list("alternative"="two.sided"),
                      step_increase = 0.2,
                      vjust = 0.2)
      

p<- a +  b  + plot_layout(nrow = 1,ncol = 2)

ggsave("results/GDSC_cancerxgene_EHMTvsHDAC.pdf",p,height = 6,width=12)

print(p)
```

\newpage

# Statistical testing (CancerXGene)

In a next step we can use ANOVA on the linear model to assess the impact of the compound used (Vorinostat) and  EHMT2 expression on the area under the curve (auc). ANOVA analysis on the model will then measure how much EHMT2 expression adds to the prediction of the auc. 

ANOVA of EHMT2 influence on drug auc unfiltered quantitative data

```{r anova, echo=FALSE}

joined_screening_data_CRC %>% 
     filter(drug_name %in% c("Vorinostat")) %>% 
  lm(formula = auc~EHMT2) %>% 
  summary()

```

We can observe significant p-value (<0.05) for EHTM2 as a predictive variable for Vorinostat sensitivity. This means that EHTM2 expression predicts HDAC inhibitor sensitivity. From the graphics and the direction of the estimate we can further conclude that there is a positive correlation between EHTM expression and HDAC inhibitor efficacy (auc), the higher EHMT expression, the less potent HDAC inhibitors are.

The same trends are found and even more pronounced when just comparing the cell lines with the highest and the lowest quantile of EHMT2 expression.

\newpage

#Testing of EHMT2 on filtered stratified data

EHTM2, Vorinostat

```{r filtered_anova_EHTM2_Vorinostat, echo=FALSE}

data_to_plot %>% 
  filter(EHMT2q!="medium",drug_name=="Vorinostat")  %>%
  t.test(formula = auc~EHMT2q,data = .) %>%
  print()


```

\newpage

# Validate on PRISM data set

Next we validate our observations on a secondary drug screening dataset.

First, we get the conversion of Cosmic to CCLE to PRISM identifiers for cell lines

```{r, CCLE_cell_line_mapping, echo=T,warning=F,error=F,message=F}
#rename mapping columns so they can be joined with others
CCLE_line_mapping <- read_delim("source_data/sample_info.csv",delim = ",") %>%
  dplyr::select(ccle_id=`CCLE Name`,cosmic_sample_id=COSMIC_ID,lineage) %>% 
  mutate(cosmic_sample_id=as.character(cosmic_sample_id))

```

Second, we get the PRISM data

```{r PRISM, echo=T,warning=F,error=F,message=F}

PRISM_cell_line <- 
  read_delim("source_data/primary-screen-cell-line-info.csv",delim = ",")

PRISM_drug <- 
  read_delim("source_data/primary-screen-replicate-collapsed-treatment-info.csv"
             ,delim = ",")
    
PRISM_data <- 
  read_delim("source_data/primary-screen-replicate-collapsed-logfold-change.csv",
            delim = ",") %>% 
  dplyr::rename(row_name=X1) %>% 
  gather(col_name,sensitivity,-row_name)

drugs_of_interest <- 
  PRISM_drug %>% 
  filter(grepl("vorinostat",name)) %>% 
  dplyr::select(col_name=column_name,name)

cells_of_interest <- PRISM_cell_line %>% 
  filter(grepl("color",primary_tissue)) %>% 
  dplyr::select(row_name,ccle_id=ccle_name) %>% 
  left_join(CCLE_line_mapping)

PRISM_data_filtered_anno_join <-  PRISM_data %>%
  filter(col_name %in% drugs_of_interest$col_name) %>%
  filter(row_name %in% cells_of_interest$row_name) %>%
  left_join(cells_of_interest) %>%
  left_join(cell_expression_data) %>%
  left_join(drugs_of_interest) %>%
  drop_na()

data_to_plot <- PRISM_data_filtered_anno_join %>%
    dplyr::select(sensitivity,ccle_id,gene_symbols,normed_expression,name) %>%
    spread(gene_symbols,normed_expression) %>%
    mutate(EHMT2q = if_else(EHMT2>quantile(EHMT2,probs = 0.75,na.rm = T),"high", 
    if_else(EHMT2<quantile(EHMT2,probs = 0.25,na.rm = T),"low","medium"))) %>%
    mutate(EHMT2q=factor(EHMT2q,levels = c("low","medium","high"))) 

```

\newpage

# Plot the trend between expression and sensitivity (PRISM)

```{r EHMT_vs_PRISM_plot, echo=FALSE,warning=F,error=F,message=F}

p1 <- data_to_plot %>%
  ggplot(aes(x=EHMT2,y=sensitivity)) +
    geom_point(aes(col=EHMT2q)) +
    geom_smooth(method = "lm") +
          theme_b110()+ 
          stat_cor(method = "pearson", label.x = 4, label.y = 1) + 
          scale_color_manual(values = c(google_blue,b110_grey_light,google_red))

give.n <- function(x){
  #return(c(y = max(abs(x))*1.5, label =length(x)))
  return(c(y = 2.5, label =length(x)))
  # experiment with the multiplier to find the perfect position
}

p2 <- data_to_plot %>%
      filter(EHMT2q != "medium") %>%
      ggplot(aes(x = EHMT2q,y = sensitivity)) + 
        geom_boxplot(aes(fill = EHMT2q)) + 
          theme_b110()+
          scale_fill_manual(values = c(google_blue,google_red)) +
          stat_summary( fun.data = give.n, 
                        geom = "text", 
                       # fun.y = median,
                        position = position_dodge(width = 0.75))+
          geom_signif(na.rm=T,comparisons = list(c("low","high")),test = "t.test",test.args = list("alternative"="two.sided"),step_increase = 0.2,vjust = 0.2)

p<- p1 + p2 + plot_layout(ncol = 2) 

ggsave("results/PRISM_EHMTvsHDAC.pdf",p,height = 6,width=12)

print(p)
```


# ANOVA trend between expression and sensitivity (PRISM)

```{r EHMT_vs_PRISM_stats,echo=FALSE}

data_to_plot %>%
  lm(sensitivity~EHMT2,data=.) %>%
  summary()
```


# Statistical testing of separate combinations

Vorinostat EHMT2 separate (t-test)

```{r, echo=FALSE}

data_to_plot %>%
  filter(EHMT2q!="medium",name=="vorinostat") %>%
  t.test(sensitivity~EHMT2q,data=.) %>%
  print()
 
```

\newpage
# Session info

```{r}
writeLines(capture.output(sessionInfo()), paste("results/SessionInfo.txt",sep = ""))
sessionInfo()
```

\newpage
# References
